### PR TITLE
NOD | Redirect to v2 questions

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > When a Veteran has the NOD v2 feature toggle enabled, they could be well past the new NOD v2 questions located near the beginning of the form. This PR uses the save-in-progress migration to redirect these Veterans back to the request extension question when their feature toggle is enabled.
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > - Redirect based on SiP data version & feature toggle
  > - Because migrations are based on the version number, the function is only added to the migration list if the feature toggle is enabled
  > - Added info alert to inform Veteran of why they were redirected
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  > nod_part3_update

## Related issue(s)

- [#67645](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67645)
- [Design](https://www.sketch.com/s/0424c51d-7818-43ea-8821-6de91eb7ab27/a/JAWb2a5)

## Testing done

- _Describe what the old behavior was prior to the change_
  > No redirect
- _Describe the steps required to verify your changes are working as expected_
  > - Log into any user
  > - Navigation to [Form 10182](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182)
  > - Turn off feature toggle (`nod_part3_update`)
  > - Start NOD and progress to the contestable issue page, or after
  > - Use finish this application later
  > - Turn on feature toggle (`nod_part3_update`)
  > - Reload the NOD application
  > - Use "Continue your request"
  > - You should now be redirected to the extension request page with info alert showing
- _Describe the tests completed and the results_
  > Added unit tests for migration & other changes; all passing
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         |  |
| ------- | ------ | 
| Request extension after redirect  | Screenshot 2023-10-16 at 1 35 48 PM |
| Request extension showing Voiceover | Screenshot 2023-10-16 at 4 00 55 PM |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/10182/tests/migrations/02-redirect-to-v2-quetions.unit.spec.js
Comment (anniebtran) [inline src/applications/appeals/10182/tests/migrations/02-redirect-to-v2-quetions.unit.spec.js:1]: just a small typo in the file name — `quetions` => `questions`

**Review by anniebtran — COMMENTED** [review]
i still wanna try pulling this down and trying it out locally, but wanted to at least submit a comment about a small nit

File: src/applications/appeals/10182/tests/migrations/02-redirect-to-v2-quetions.unit.spec.js
Comment (Mottie) [reply src/applications/appeals/10182/tests/migrations/02-redirect-to-v2-quetions.unit.spec.js:1]: Thanks for catching this. Fixed!

**Review by anniebtran — APPROVED** [review]
tested locally per your instructions and it looks good! :shipit:

**Pull request merged**